### PR TITLE
Provide Func<ReadOnlySettings, IDocumentStore> configuration options

### PR DESCRIPTION
**Pull request by DavidBoike**
Connects to #214

Related to Particular/PlatformDevelopment#733

### CODE REVIEW

**Comment by DavidBoike** [comment]
@Particular/ravendb-persistence-maintainers ready for review. Note this adds public API.

**Comment by SimonCropp** [comment]
Matching doco pr?

**Comment by SimonCropp** [comment]
Is this missing some argument null checks?

**Comment by DavidBoike** [comment]
An item for a doco PR for this change is in the PoA for Particular/PlatformDevelopment#733. Didn't have time for it today but I can tackle it tomorrow, provided nobody comes up with a reason why adding this API would be the worst idea ever.

File: src/NServiceBus.RavenDB/Timeouts/RavenDbTimeoutSettingsExtensions.cs [hunk 1]
```diff
@@ -15,10 +17,20 @@ public static class RavenDbTimeoutSettingsExtensions
...
+        }
+
+        /// <summary>
+        ///     Configures the given document store to be used when storing timeouts
+        /// </summary>
+        /// <param name="cfg"></param>
+        /// <param name="storeCreator">A Func that will create the document store on NServiceBus initialization.</param>
+        public static PersistenceExtentions<RavenDBPersistence> UseDocumentStoreForTimeouts(this PersistenceExtentions<RavenDBPersistence> cfg, Func<ReadOnlySettings, IDocumentStore> storeCreator)
+        {
+            DocumentStoreManager.SetDocumentStore<StorageType.Timeouts>(cfg.GetSettings(), storeCreator);
```
Comment (mauroservienti) [inline src/NServiceBus.RavenDB/Timeouts/RavenDbTimeoutSettingsExtensions.cs]: any value in checking against `null` incoming `storeCreator`?

**Comment by mauroservienti** [comment]
As @SimonCropp pointed out other than some possible missing null guards, I cannot spot anything so this looks good and can be merged.

**Comment by DavidBoike** [comment]
Will add the null guards today, and (if there are no objections) self-merge after that, and get to work on the doco as well.

File: src/NServiceBus.RavenDB/Timeouts/RavenDbTimeoutSettingsExtensions.cs (hunk 1 above)
Comment (DavidBoike) [inline src/NServiceBus.RavenDB/Timeouts/RavenDbTimeoutSettingsExtensions.cs]: Added in the DocumentStoreManager methods for centralization.

**Pull request merged**